packages/ipfs-core/src/components/storage.js

```diff
@@ -43,6 +43,8 @@ class Storage {
 
     const { peerId, keychain, isNew } = await loadRepo(repo, options)
 
+    // TODO: throw error?
```

Comment:
Keychain is not available in init IIRC. That is why it might exist or not

Reply:
So there was / is this code:

https://github.com/ipfs/js-ipfs/blob/2bafc884da6f1153f8ab2eea4f2418b372ca80e0/packages/ipfs-core/src/components/storage.js#L137-L156

And this code path

https://github.com/ipfs/js-ipfs/blob/2bafc884da6f1153f8ab2eea4f2418b372ca80e0/packages/ipfs-core/src/components/storage.js#L214-L230

And I'm still not sure why libp2p is created in two different ways to be honest. It would be good to get these cleaned up (in some followup). I'd also happily pair on this as lack of uncertainty here really bothered me.

Reply:
I agree with you on this, it also confused me. We should improve this on a followup